File: frameworks/domain.core/src/test/java/com/qomolangma/frameworks/domain/core/UuidBasedIdGeneratorTest.java
package com.qomolangma.frameworks.domain.core;

import org.junit.jupiter.api.*;

import static org.assertj.core.api.Assertions.assertThat;

public class UuidBasedIdGeneratorTest {

    private Id id;

    @BeforeEach
    public void setUp() {
        id = new UuidBasedId();
    }

    @Test
    public void should_generate_identity_correctly() {
        assertThat(32).isEqualTo(id.next().length());
    }

    @Test
    public void should_generate_non_duplicated_identities() {
        String id = this.id.next();
        assertThat(id).isNotEqualTo(this.id.next());
    }

    @Test
    public void should_generate_readable_identity_correctly() {
        assertThat(id.nextReadable().length()).isEqualTo(21);
        assertThat(id.nextReadable().chars().allMatch(Character::isDigit)).isTrue();
    }

    @Test
    public void should_generate_non_duplicated_readable_identities() {
        String id = this.id.nextReadable();
        assertThat(this.id.nextReadable()).isNotEqualTo(id);
    }

}

File: frameworks/domain.core/src/test/kotlin/com/qomolangma/frameworks/domain/core/UuidBasedIdGeneratorTest.kt
package com.qomolangma.frameworks.domain.core

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class UuidBasedIdGeneratorTest {
    private var idGenerator: IdGenerator? = null

    @BeforeEach
    fun setUp() {
        idGenerator = UuidBasedIdGenerator()
    }

    @Test
    fun should_generate_identity_correctly() {
        Assertions.assertThat(32).isEqualTo(idGenerator!!.next().length)
    }

    @Test
    fun should_generate_non_duplicated_identities() {
        val id = idGenerator!!.next()
        Assertions.assertThat(id).isNotEqualTo(this.idGenerator!!.next())
    }

    @Test
    fun should_generate_readable_identity_correctly() {
        Assertions.assertThat(idGenerator!!.nextReadable().length).isEqualTo(21)
        Assertions.assertThat(
            idGenerator!!.nextReadable().chars().allMatch(Character::isDigit)
        ).isTrue
    }

    @Test
    fun should_generate_non_duplicated_readable_identities() {
        val id = idGenerator!!.nextReadable()
        Assertions.assertThat(this.idGenerator!!.nextReadable()).isNotEqualTo(id)
    }
}
